src/rte/mo_rte_sw.jl

```diff
@@ -136,95 +162,101 @@ contains
     #   On input flux_dn is the diffuse component; the last action in each solver is to add
     #   direct and diffuse to represent the total, consistent with the LW
     #
-    #$acc enter data copyin(mu0)
-    #$acc enter data create(gpt_flux_up, gpt_flux_dn, gpt_flux_dir)
-
-    #$acc enter data copyin(inc_flux)
-    call apply_BC(ncol, nlay, ngpt, logical(top_at_1, wl),   inc_flux, mu0, gpt_flux_dir)
-    #$acc exit data delete(inc_flux)
-    if(present(inc_flux_dif)) then
-      #$acc enter data copyin(inc_flux_dif)
-      call apply_BC(ncol, nlay, ngpt, logical(top_at_1, wl), inc_flux_dif,  gpt_flux_dn )
-      #$acc exit data delete(inc_flux_dif)
-    else
-      call apply_BC(ncol, nlay, ngpt, logical(top_at_1, wl),                gpt_flux_dn )
-    end if
+#    #$acc enter data copyin(mu0)
+#    #$acc enter data create(gpt_flux_up, gpt_flux_dn, gpt_flux_dir)
 
-    select type (atmos)
-      class is (ty_optical_props_1scl)
+#    #$acc enter data copyin(inc_flux)
+    apply_BC(ncol, nlay, ngpt, logical(top_at_1, wl),   inc_flux, mu0, gpt_flux_dir)
+#    #$acc exit data delete(inc_flux)
+    if(present(inc_flux_dif)) #then
+#      #$acc enter data copyin(inc_flux_dif)
+      apply_BC(ncol, nlay, ngpt, logical(top_at_1, wl), inc_flux_dif,  gpt_flux_dn )
+#      #$acc exit data delete(inc_flux_dif)
+    else
+      apply_BC(ncol, nlay, ngpt, logical(top_at_1, wl),                gpt_flux_dn )
+    end
+#    select type (atmos)
+      if ( atmos isa ty_optical_props_1scl )
         #
         # Direct beam only
         #
-        #$acc enter data copyin(atmos, atmos%tau)
-        error_msg =  atmos%validate()
-        if(len_trim(error_msg) > 0) return
-        call sw_solver_noscat(ncol, nlay, ngpt, logical(top_at_1, wl), &
-                              atmos%tau, mu0,                          &
+#        #$acc enter data copyin(atmos, atmos%tau)
+        error_msg =  validate(atmos)
+        if(length(strip(error_msg)) > 0) 
+	  return
```

Comment:
```suggestion
	  return error_msg
```
I would add this just in case